distributed/tests/test_priorities.py

```diff
@@ -1,128 +1,232 @@
+from __future__ import annotations
+
 import asyncio
+from contextlib import asynccontextmanager
 
 import pytest
 
 import dask
-from dask import delayed, persist
-from dask.core import flatten
-from dask.utils import stringify
-
-from distributed import Worker, wait
-from distributed.utils_test import gen_cluster, inc, slowdec, slowinc
-
-
-@gen_cluster(client=True, nthreads=[])
-async def test_submit(c, s):
-    low = c.submit(inc, 1, priority=-1)
-    futures = c.map(slowinc, range(10), delay=0.1)
-    high = c.submit(inc, 2, priority=1)
-    async with Worker(s.address, nthreads=1):
-        await wait(high)
-        assert all(s.processing.values())
-        assert s.tasks[low.key].state == "processing"
-
-
-@gen_cluster(client=True, nthreads=[])
-async def test_map(c, s):
-    low = c.map(inc, [1, 2, 3], priority=-1)
-    futures = c.map(slowinc, range(10), delay=0.1)
-    high = c.map(inc, [4, 5, 6], priority=1)
-    async with Worker(s.address, nthreads=1):
-        await wait(high)
-        assert all(s.processing.values())
-        assert s.tasks[low[0].key].state == "processing"
-
-
-@gen_cluster(client=True, nthreads=[])
-async def test_compute(c, s):
-    da = pytest.importorskip("dask.array")
-    x = da.random.random((10, 10), chunks=(5, 5))
-    y = da.random.random((10, 10), chunks=(5, 5))
-
-    low = c.compute(x, priority=-1)
-    futures = c.map(slowinc, range(10), delay=0.1)
-    high = c.compute(y, priority=1)
-    async with Worker(s.address, nthreads=1):
-        await wait(high)
-        assert all(s.processing.values())
-        assert s.tasks[stringify(low.key)].state in ("processing", "waiting")
-
-
-@gen_cluster(client=True, nthreads=[])
-async def test_persist(c, s):
-    da = pytest.importorskip("dask.array")
-    x = da.random.random((10, 10), chunks=(5, 5))
-    y = da.random.random((10, 10), chunks=(5, 5))
-
-    low = x.persist(priority=-1)
-    futures = c.map(slowinc, range(10), delay=0.1)
-    high = y.persist(priority=1)
-    async with Worker(s.address, nthreads=1):
-        await wait(high)
-        assert all(s.processing.values())
-        assert all(
-            s.tasks[stringify(k)].state in ("processing", "waiting")
-            for k in flatten(low.__dask_keys__())
-        )
-
-
-@gen_cluster(client=True)
-async def test_annotate_compute(c, s, a, b):
+from dask import delayed
+
+from distributed import Client, Event, Scheduler, Status, Worker, wait
+from distributed.utils_test import gen_cluster, inc, slowinc
+
+dinc = delayed(inc)
+dslowinc = delayed(slowinc)
+dwait = delayed(lambda ev: ev.wait())
+
+
+@asynccontextmanager
+async def block_worker(
+    c: Client,
+    s: Scheduler,
+    w: Worker,
+    pause: bool,
+    ntasks_on_scheduler: int | None = None,
+    ntasks_on_worker: int | None = None,
```

Comment:
We're not using these anywhere, are we? Why expose this as part of the signature?

Reply:
we're using it in the last 2 tests